backend/core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueService.java

```diff
@@ -429,14 +431,21 @@ public static RaisedIssueDto raisedIssueUpdater(RaisedIssueDto previouslyRaisedI
       UnaryOperator<RaisedIssueDto> issueUpdater = it -> it.builder().withType(RuleType.valueOf(userType.name())).buildIssue();
       updatedIssue = updateIssue(updatedIssue, impactedIssueKeys, issueUpdater);
     }
-    // TODO: Adapt with CCA/Impacts from the event
-    UnaryOperator<RaisedIssueDto> issueUpdater;
-    if (isMQRMode) {
-      issueUpdater = it -> it.builder().withMQRModeDetails(it.getCleanCodeAttribute(), it.getImpacts()).buildIssue();
-    } else {
-      issueUpdater = it -> it.builder().withStandardModeDetails(it.getSeverity(), it.getType()).buildIssue();
+    for (var issue : event.getImpactedIssues()) {
+      if (!issue.getImpacts().isEmpty() && isMQRMode) {
+        var impacts = issue.getImpacts().entrySet().stream()
+          .map(impact ->
+            new ImpactDto(
+              SoftwareQuality.valueOf(impact.getKey().name()),
+              org.sonarsource.sonarlint.core.rpc.protocol.common.ImpactSeverity.valueOf(impact.getValue().name())
+            )
+          ).collect(Collectors.toList());
+        UnaryOperator<RaisedIssueDto> issueUpdater =
+          it -> it.builder().withMQRModeDetails(it.getCleanCodeAttribute(), impacts).buildIssue();
```

Comment:
Same thing here, you might be replacing the full list of impacts with just the overridden ones from the event. You probably need to merge what comes from the event with the previously raised issue